### PR TITLE
Inserting proof notes

### LINKED ISSUE #178: Document proof structure
This is the second of four issues on the addition into the notes files. We need to add the procedure of constructing a general SNARK aswell as constructing a proof for PLONK.

### PR DESCRIPTION
**Pull request by Bounce23**
This PR is for porting the notes on generic SNARK construction and specific PLONK proof construction from `docs`,  for release.

This closes #178

### CODE REVIEW

File: docs/notes-prove-verify.md [hunk 1]
```diff
@@ -1,6 +1,392 @@
...
+like expressions. Using the same
+example of:
+
+\\[
+\mathbf{W}\_*L* \cdot \mathbf{W}\_*R* = \mathbf{W}\_*O*,
+\\]
+
+We can express multiples of the 
+same wires in or out of the same 
+gate, with the above equation.
```
Comment (CPerezz) [inline docs/notes-prove-verify.md:30]: Multiples of  `the same wires`? WTYM?

File: docs/notes-prove-verify.md [hunk 2]
```diff
@@ -1,6 +1,392 @@
...
+\\]
+
+PLONK also uses 'copy
+constraints', which are used to 
+associate wires, which have 
+equality, from the entire circuit.
+These constriants are checked 
+with a permutation argument. 
+In essence, this checks that 
+wires are mot repeated by using
```
Comment (CPerezz) [inline docs/notes-prove-verify.md:74]: typo: `not`

File: docs/notes-prove-verify.md [hunk 3]
```diff
@@ -1,6 +1,392 @@
...
+system of mumerical equations, 
+which in PLONK are reduced to a 
+small amount of polynomial 
+equations which are capable of 
+representing the two types.
+PLONK allows us to combine the
+two gate equations by describing 
+their relationship relative to 
+the role in the circuit.
+PLONK also has constant, which
```
Comment (CPerezz) [inline docs/notes-prove-verify.md:91]: `a constant`. Anyway, we have much more constants, (If we understand that constant = selector poly coeff)

Comment (CPerezz) [reply docs/notes-prove-verify.md:91]: Otherways, you meant `constants` in which case you're missing the `s`

File: docs/notes-prove-verify.md [hunk 4]
```diff
@@ -1,6 +1,392 @@
...
+are denoted as 'Q'. These 
+values will change for each 
+programme. When they are
```
Comment (CPerezz) [inline docs/notes-prove-verify.md:94]: We should mention here that they are actually defining the circuit "shape" and have nothing to do with the witness values

File: docs/notes-prove-verify.md [hunk 5]
```diff
@@ -1,6 +1,392 @@
...
+to convert coefificient form. 
+All this interpolation is doing
+is allowing us to evaluate a
+functions over specific points,
+for 'x' values,the polynomial 
+is equal to '1' or '0'.
+
+With these specific basis, we
+can derive the relation between 
+all sets fo equations into one
```
Comment (CPerezz) [inline docs/notes-prove-verify.md:188]: typo -> `of`

File: docs/notes-prove-verify.md [hunk 6]
```diff
@@ -1,6 +1,392 @@
...
+based on interpolation of two 
+functions of a particular group
+elements. The following section 
+gives a more comprehensive
+understanding to the way in 
+which these polynomials are 
+formed, given certain inputs. 
+
+
+Langrangian polynomials are
```
Comment (CPerezz) [inline docs/notes-prove-verify.md:262]: Shouldn't we add a tittle or something similar?

File: docs/notes-prove-verify.md (hunk 1 above)
Comment (Bounce23) [reply docs/notes-prove-verify.md:30]: copy constraints is what it means. I can change it to be more clear

File: docs/notes-prove-verify.md (hunk 3 above)
Comment (Bounce23) [reply docs/notes-prove-verify.md:91]: I'll change the typo but when the constants are expressed, it's still the vectors expressed as one term in the polynomial

File: docs/notes-prove-verify.md (hunk 4 above)
Comment (Bounce23) [reply docs/notes-prove-verify.md:94]: Will add in.

File: docs/notes-prove-verify.md (hunk 6 above)
Comment (Bounce23) [reply docs/notes-prove-verify.md:262]: Well, this is the separate part, where we add in possibly else where. Even still, you're right, i'll add one :)

File: docs/notes-prove-verify.md (hunk 1 above)
Comment (CPerezz) [reply docs/notes-prove-verify.md:30]: ohh. Fine. Whatever you think it's more clear.

File: docs/notes-prove-verify.md (hunk 3 above)
Comment (CPerezz) [reply docs/notes-prove-verify.md:91]: ack

**Pull request merged**